common:
    lid_topic: "/livox/lidar"
    imu_topic: "/livox/imu"
    time_sync_en: false # ONLY turn on when external time synchronization is really not possible

preprocess:
    lidar_type: 1 # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
    scan_line: 4
    blind: 0.5

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree: 360
    det_range: 100.0
    extrinsic_est_en: false # true: enable the online estimation of IMU-LiDAR extrinsic
    extrinsic_T: [-0.011, -0.02329, 0.04412]
    extrinsic_R: [1, 0, 0, 0, 1, 0, 0, 0, 1]

publish:
    path_publish_en: true
    scan_publish_en: true # false: close all the point cloud output
    dense_publish_en: false # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: false # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: true
    interval:
        -1 # how many LiDAR frames saved in each pcd file;
        # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
feature_extract_enable: false
point_filter_num: 1
max_iteration: 3
filter_size_surf: 0.1
filter_size_map: 0.3 # 暂时未用到，代码中为0， 即倾向于将降采样后的scan中的所有点加入map
cube_side_length: 1000

esti_plane_threshold: 0.1 # default=0.1

localization_param:
    map_file_name: "origin_map"
    init_height: 0.2

relocalization_param:
    leaf_size: 0.5
    det_range: 50.0
    max_correspondence_distance: 3.0

manager_param:
    block_size: 1.0
    block_resolution: 0.2
    temporary_map_points_threshold: 20
    global_map_filter_leaf_size: 0.3
    search_radius: 0.5
    min_search_points_num: 3
